--- CMakeLists.txt.orig	2026-08-09 22:43:36.781917824 +0800
+++ CMakeLists.txt	2026-08-09 23:49:03.876026759 +0800
@@ -188,21 +188,10 @@
 elseif(BUILD_WASI MATCHES "metawasi")
   set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Dd_m3HasMetaWASI")
 elseif(BUILD_WASI MATCHES "uvwasi")
-  include(FetchContent)
-  FetchContent_Declare(
-    uvwasi
-    GIT_REPOSITORY https://github.com/vshymanskyy/uvwasi.git
-    GIT_TAG b063d686848c32a26119513056874f051c74258a
-  )
+    find_package(uvwasi REQUIRED CONFIG)
 
-  FetchContent_GetProperties(uvwasi)
-  if(NOT uvwasi_POPULATED)
-    FetchContent_Populate(uvwasi)
-    include_directories("${uvwasi_SOURCE_DIR}/include")
-    add_subdirectory(${uvwasi_SOURCE_DIR} ${uvwasi_BINARY_DIR} EXCLUDE_FROM_ALL)
-  endif()
   set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Dd_m3HasUVWASI")
-  target_link_libraries(${OUT_FILE} uvwasi_a uv_a)
+  target_link_libraries(${OUT_FILE} uvwasi::uvwasi_a libuv::uv_a)
 endif()
 
 check_ipo_supported(RESULT result)
